internal/functions/registernotification/register-notification.go

```diff
@@ -48,9 +49,42 @@ func RegisterNotification(w http.ResponseWriter, r *http.Request) {
 
 	logger.Debugf("Handling RegisterNotification request: %v %+v", ehrid, request)
 
+	isEhrid, _ := regexp.MatchString(utils.EhridRegex, ehrid)
+
+	if !isEhrid {
+		logger.Infof("Provided ID is not eHrid: %v", ehrid)
+		err = handleForFUID(ctx, ehrid)
+	} else {
+		err = handleForEhrid(ctx, ehrid)
+	}
```

Comment:
maybe we could rename the variable ehrid to something like `token` since we don't know if its ehrid or fuid? :D

Reply:
Fair point. It's `UID` in fact so I changed it.